tools-ui/src/main/java/com/hedera/hashgraph/client/ui/CreatePaneController.java

```diff
@@ -2213,5 +2262,199 @@ private void setContentsAction() {
 			contents = null;
 		}
 	}
+
+	public void loadFormFromTransaction() {
+		File transactionFile = null;
+		cleanFields();
+		if (SetupPhase.NORMAL_OPERATION_PHASE.equals(controller.getSetupPhase())) {
+			transactionFile = loadTransaction();
+		}
+		if (SetupPhase.TEST_PHASE.equals(controller.getSetupPhase()) && !"".equals(loadTransactionTextField.getText())) {
+			transactionFile = new File(loadTransactionTextField.getText());
+		}
+		if (transactionFile == null) {
+			return;
+		}
+		if (ZIP_EXTENSION.equals(FilenameUtils.getExtension(transactionFile.getName()))) {
+			try {
+				loadLargeFileUpdateToForm(transactionFile);
+			} catch (HederaClientException e) {
+				PopupMessage.display("Error loading file", "Cannot load information from zip file");
+			}
+			return;
+		}
+		if (!TRANSACTION_EXTENSION.equals(FilenameUtils.getExtension(transactionFile.getName()))) {
+			return;
+		}
+		ToolTransaction transaction;
+		try {
+			transaction = new ToolTransaction().parseFile(transactionFile);
+		} catch (HederaClientException e) {
+			logger.error(e.getMessage());
+			PopupMessage.display("Invalid transaction", "The file selected cannot be loaded.");
+			return;
+		}
+
+		switch (transaction.getTransactionType()) {
+			case CRYPTO_TRANSFER:
+				selectTransactionType.setValue("Transfer");
+				loadCryptoTransferToForm((ToolTransferTransaction) transaction);
+				break;
+			case CRYPTO_CREATE:
+				selectTransactionType.setValue("Account Creation");
+				loadCryptoCreateToForm((ToolCryptoCreateTransaction) transaction);
+				break;
+			case CRYPTO_UPDATE:
+				selectTransactionType.setValue("Account Update");
+				loadCryptoUpdateToForm((ToolCryptoUpdateTransaction) transaction);
+				break;
+			case SYSTEM_DELETE_UNDELETE:
+				selectTransactionType.setValue("Admin Modify Content");
+				loadSystemTransactionToForm((ToolSystemTransaction) transaction);
+				break;
+			case FILE_UPDATE:
+			case FILE_APPEND:
```

Comment:
nit: I would replace this two with a `default` since there are lots of other possible transaction types are not handled here, unless there are special considerations here.

Reply:
Originally I intended to implement those missing transactions, but I ran out of time. I will replace them with a `default` branch as you suggest and when I revisit this part of the code add each branch.

Reply:
Understood. Just thought a `switch` without a `default` should be very carefully treated.